autoPyTorch/api/base_task.py

```diff
@@ -1265,21 +1507,102 @@ def fit(self,
-            dataset=dataset,
-            split_id=split_id)
+        pipeline_options = self.pipeline_options.copy().update(pipeline_options) if pipeline_options is not None \
+            else self.pipeline_options.copy()
 
-        fit_and_suppress_warnings(self._logger, pipeline, X, y=None)
+        assert pipeline_options is not None
 
+        if budget_type is not None:
+            pipeline_options.update({'budget_type': budget_type})
+        else:
+            budget_type = pipeline_options['budget_type']
+
+        budget = budget if budget is not None else pipeline_options[budget_type]
+
+        if disable_file_output is None:
+            disable_file_output = self._disable_file_output if hasattr(self, '_disable_file_output') \
+                and self._disable_file_output is not None else []
+
+        stats.start_timing()
+
+        tae = ExecuteTaFuncWithQueue(
+            backend=self._backend,
+            seed=self.seed,
+            metric=metric,
+            logger_port=self._logger_port,
+            cost_for_crash=get_cost_of_crash(metric),
+            abort_on_first_run_crash=False,
+            initial_num_run=self._backend.get_next_num_run(),
+            stats=stats,
+            memory_limit=memory_limit,
+            disable_file_output=disable_file_output,
+            all_supported_metrics=all_supported_metrics,
+            budget_type=budget_type,
+            include=include_components,
+            exclude=exclude_components,
+            search_space_updates=search_space_updates,
+            pipeline_config=pipeline_options,
+            pynisher_context=self._multiprocessing_context
+        )
+
+        run_info, run_value = tae.run_wrapper(
+            RunInfo(config=configuration,
+                    budget=budget,
+                    seed=self.seed,
+                    cutoff=run_time_limit_secs,
+                    capped=False,
+                    instance_specific=None,
+                    instance=None)
+        )
+
+        fitted_pipeline: Optional[BasePipeline] = None
+
+        if run_value.status == StatusType.SUCCESS:
+            if 'all' in disable_file_output or 'pipeline' in disable_file_output:
+                self._logger.warning("File output is disabled. No pipeline can returned")
+            else:
+                if self.resampling_strategy in CrossValTypes:
+                    load_function = self._backend.load_cv_model_by_seed_and_id_and_budget
+                else:
+                    load_function = self._backend.load_model_by_seed_and_id_and_budget
+                fitted_pipeline = load_function(
+                    seed=self.seed,
+                    idx=run_info.config.config_id + tae.initial_num_run,
+                    budget=float(run_info.budget),
+                )
+        else:
+            warnings.warn(f"Fitting pipeline failed with status: {run_value.status}"
+                          f", additional_info: {run_value.additional_info}")
```

Comment:
Can we transplant?
Where should `self._clean_logger()` come? you can include it here if you think you need it here.
```python
    def _get_fitted_pipeline(...) -> ...:
        if run_value.status != StatusType.SUCCESS:
            warnings.warn(f"Fitting pipeline failed with status: {run_value.status}"
                          f", additional_info: {run_value.additional_info}")
            return None
        elif any(disable_file_output for c in ['all', 'pipeline']):
            self._logger.warning("File output is disabled. No pipeline can returned")
            return None

        if self.resampling_strategy in CrossValTypes:
            load_function = self._backend.load_cv_model_by_seed_and_id_and_budget
        else:
            load_function = self._backend.load_model_by_seed_and_id_and_budget

        return load_function(
            seed=self.seed,
            idx=run_info.config.config_id + tae.initial_num_run,
            budget=float(run_info.budget),
        )

```

Reply:
sure.